### DISCUSSION TITLE
Why not use gRPC instead of web API

### DISCUSSION THREAD

**Discussion by lthoa** [oqtane/oqtane.framework#972] [Ideas]
I think gRPC is really good with [code-first support](https://github.com/protobuf-net/protobuf-net.Grpc). Why not use gRPC instead of web API?

**Comment by jimspillane** [comment]
I agree, and after [gRPC services with C#](https://docs.microsoft.com/en-us/aspnet/core/grpc/basics?view=aspnetcore-5.0) is supported on Azure App Service and IIS, we should consider adding it to Oqtane.

![image](https://user-images.githubusercontent.com/15007083/101078319-00c01980-3574-11eb-8615-6f95ca0b0cb1.png)

**Comment by sbwalker** [comment]
Developers should be able to use gRPC in their own custom modules - they are not restricted by the technologies used in the core framework.

**Comment by sbwalker** [comment]
Oqtane contains a headless API which allows different clients to interact with it. By default it has a Blazor client but there is no reason why someone could not create a client using Angular or React or ???. In addition, the API has Swagger support by default and there is the ability to interact with it using tools like curl or Postman. As a result the API needs to be based on technology which has the broadest industry adoption and support. At this stage I do not believe gRPC fits this category.
